doc/user-guide.md

```diff
@@ -423,6 +423,62 @@ To implement complex deletion logic, you can add a finalizer to your Custom Reso
 deleted until you remove the finalizer (ie, after your cleanup logic has successfully run). For more information, see the
 [official Kubernetes documentation on finalizers](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#finalizers).
 
+**Example:**
+
+Following is a snippet from the controller file under `pkg/controller/memcached/memcached_controller.go`
+
+```go
+func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+    reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+    reqLogger.Info("Reconciling Memcached")
+        
+    // Fetch the Memcached instance
+    memcached := &cachev1alpha1.Memcached{}
+    err := r.client.Get(context.TODO(), request.NamespacedName, memcached)
+    
+	// Check if the APP CR was marked to be deleted
+	isCRMarkedToBeDeleted := memcached.GetDeletionTimestamp() != nil
+	if isCRMarkedToBeDeleted {
+		// TODO(user): Add the impl that is required to allow the CR be deleted
+
+		// Update finalizer to allow delete CR
+		memcached.SetFinalizers(nil)
+
+		// Update CR
+		err := r.client.Update(context.TODO(), memcached)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{}, nil
+	}
+	
+	// Add finalizer for this CR
```

Comment:
The indentation for Lines 440-455 are off. I think they should be aligned with  the rest of the function to just 1 tab space.